lib/elliptic/ec/index.js

```diff
@@ -164,3 +164,45 @@ EC.prototype.verify = function verify(msg, signature, key, enc) {
 
   return p.getX().mod(this.n).cmp(r) === 0;
 };
+
+EC.prototype.recoverPubKey = function(msg, signature, j) {
+
+  assert(3 & j === j, 'The recovery param is more than two bits');
+  var n = this.n;
+  var e = new bn(msg);
+  var r = signature.r;
+  var s = signature.s;
+
+  // A set LSB signifies that the y-coordinate is odd
+  var isYOdd = j & 1;
+
+  // The more significant bit specifies whether we should use the
+  // first or second candidate key.
+  var isSecondKey = j >> 1;
+
+  // 1.1. Let x = r + jn.
+  var x = isSecondKey ? n.add(r) : r;
+  r = this.curve.pointFromX(isYOdd, x);
+
+  var rn = r.mul(n);
+  assert(rn.isInfinity(), 'nR is not a valid curve point');
+
+  var eNeg = e.neg().mod(n);
+
+  // 1.6.1 Compute Q = r^-1 (sR -  eG)
+  //               Q = r^-1 (sR + -eG)
+  var rInv = signature.r.invm(n);
+
+  var Q = r.mul(s).add(this.g.mul(eNeg)).mul(rInv);
+  assert(this.curve.validate(Q), 'Invalid Point');
+  return Q;
+};
+
+EC.prototype.calcPubKeyRecoveryParam = function(e, signature, Q) {
+  for (var i = 0; i < 4; i++) {
+    var Qprime = this.recoverPubKey(e, signature, i);
+
+    if (Qprime.eq(Q))  return i;
+  }
+  throw new Error('Unable to find valid recovery factor');
```

Comment:
This is slow. But I don't know the correct way to do this. Bitcoin generates the recovery ID on when signing. See [here](https://github.com/bitcoin/secp256k1/blob/master/src/ecdsa_impl.h#L223-L260) I'm not sure what the `overflow` is doing yet.  So I haven't attempted at porting it.